### DISCUSSION TITLE
No model in devicetree_unfixed.h :

### DISCUSSION THREAD

**Discussion by yfourel-vossloh** [zephyrproject-rtos/zephyr#48278] [Q&A]
I use zephyr 3.0 on a STM32H743.

We add "model" in our zephyr.dts file because we need to send model name during Mqtt Nbirth request.
Here is the beginning of zephyr.dts : 

```c
/dts-v1/;

/ {
	#address-cells = < 0x1 >;
	#size-cells = < 0x1 >;
	model = "VOG ETH board";
	compatible = "vossloh,vog-eth";
	chosen {
		zephyr,entropy = &rng;
		zephyr,flash-controller = &flash;
		zephyr,console = &lpuart1;
```

- My problem is that I can find "compatible" on devicetree_unfixed.h, but not "model". I think it's a bug because you allow to add model property, but it seems it's not generated in devicetree_unfixed.h
- Moreover, is it possible to access root property ? No result for #define MODEL_TREE  DT_PROP(DT_ROOT, compatible), whereas I can access nodes properties.

**Comment by gmarull** [comment]
Does `vossloh,vog-eth` binding exist with `model` property? `model` should likely be part of `base.yaml`.

**Reply by yfourel-vossloh** [reply]
Thanks for your answer.
My colleague is in vacation during two weeks, I prefer he'll answer you because he worked on yaml and dts files more than me.

**Reply by nvincent-vossloh** [reply]
> Does `vossloh,vog-eth` binding exist with `model` property? `model` should likely be part of `base.yaml`.

The board mentionned is defined in a repository out of zephyr tree (https://docs.zephyrproject.org/3.0.0/application/index.html#custom-board-definition), `vossloh` was not defined in `dts/bindings/vendor-prefixes.txt`, however it does not change the output of `devicetree_unfixed.h`.

Please note that issue can be seen with boards defined in zephyr's repository:
```
west build -b nucleo_h743zi zephyr/samples/hello_world/
```

in `build/zephyr/zephyr.dts`:
```dts
/dts-v1/;

/ {
        #address-cells = < 0x1 >;
        #size-cells = < 0x1 >;
        model = "STMicroelectronics STM32H743ZI-NUCLEO board";
        compatible = "st,stm32h743zi-nucleo";
...
```
in `build/zephyr/include/generated/devicetree_unfixed.h`, there is no trace of `model` however there are mutliple macros related to the `compatible` property at the same level

```c
...
/* Generic property macros: */
#define DT_N_P_compatible {"st,stm32h743zi-nucleo"}
#define DT_N_P_compatible_IDX_0 "st,stm32h743zi-nucleo"
#define DT_N_P_compatible_IDX_0_EXISTS 1
#define DT_N_P_compatible_FOREACH_PROP_ELEM(fn) fn(DT_N, compatible, 0)
#define DT_N_P_compatible_FOREACH_PROP_ELEM_VARGS(fn, ...) fn(DT_N, compatible, 0, __VA_ARGS__)
#define DT_N_P_compatible_LEN 1
#define DT_N_P_compatible_EXISTS 1
...
```
